### ISSUE TITLE
Cannot find `cygpath` if it is in another directory than `bash`

### ISSUE THREAD

**Issue by jonashaag** [mamba-org/mamba#1779]
Under some specific circumstances Micro/mamba fails to find a working `cygpath` even if it exists:

- `micromamba shell init -s bash` run not from bash
- `cygpath` exists but in another directory than `bash`

In that case we are running into https://github.com/mamba-org/mamba/blob/bb69cbd7deaab7ae0405b45bdc4808a20b0acd17/libmamba/src/core/shell_init.cpp#L155 and fail to find `cygpath`.

If running `micromamba shell init -s bash` is not supported outside of bash then we should raise an error instead.

**Comment by jonashaag** [comment]
I wanted to work on this. @adriendelsalle can you help me understand this comment? What does it mean? https://github.com/mamba-org/mamba/blob/f2ac46b1c478668ed09463cf5b88d4802af7661b/libmamba/src/core/shell_init.cpp#L138-L144

**Comment by adriendelsalle** [comment]
@jonashaag sorry for the late reply.
Yes this is a very tricky one, because `cygpath` from `m2w64` would just mess the path by removing the env directory in the path entries, like a changing root.
IIRC it was done in [this PR](https://github.com/mamba-org/mamba/pull/1228).

If using `cygpath` from the env is error prone, maybe it would be a good idea to check if `cygpath` is located in the env or not when not found in the same directory as `bash`, and throw an error if it's the case?

**Comment by swenger** [comment]
@pavelzw suggested [this workaround](https://github.com/Quantco/barmer_db2_dataload_control_table_monitoring/blob/4898f055c2ab4be632d175a078ad17599375bce1/.github/workflows/schedule.yml#L18-L20) (manually adding the hard-coded path) to fix my CI job, and it seems to work.

**Comment by jonashaag** [comment]
Yep, we had a similar workaround in provision-with-micromamba I think. I'd really like to understand the root cause of this issue though.
